hathor/transaction/base_transaction.py

```diff
@@ -672,20 +672,13 @@ def get_metadata(self, *, force_reload: bool = False, use_storage: bool = True)
             #        happens include generating new mining blocks and some tests
             height = self.calculate_height() if self.storage else None
             score = self.weight if self.is_genesis else 0
-            kwargs: dict[str, Any] = {}
-
-            if self.is_block:
-                from hathor.transaction import Block
-                assert isinstance(self, Block)
-                kwargs['feature_activation_bit_counts'] = self.calculate_feature_activation_bit_counts()
```

Comment:
Why you don't have to do it anymore? Isn't this metadata used anymore?

Reply:
It's still used, but I have reorganized it in a way to try to improve it. Before, there were two methods:

1. `Block.calculate_feature_activation_bit_counts()`
2. `BaseTransaction._update_feature_activation_bit_counts_metadata()`

The metadata initialization in `BaseTransaction.get_metadata()` called the first one and stored its value in metadata. Then, `BaseTransaction.update_initial_metadata()` called the second one, which also called the first one, calculating and storing the value in metadata, too. This was implemented in analogy to the height metadata, which is calculated and saved in the same places.

There was also a third method, `Block.get_feature_activation_bit_counts()`, that simply returned the value from metadata. Having all these methods was redundant and hard to maintain. Using a similar "lazy get" strategy, I simply removed all pre-calculation of this value, and left a single method, a new `Block.get_feature_activation_bit_counts()` that calculates this value only the first time it's called, stores it in metadata, and returns the stored value in all subsequent calls.

Reply:
Got it. So you just initialize it as `None` during metadata creation, right?

Reply:
Exactly, this is done in `TransactionMetadata.__init__()`

Reply:
Thanks!